src/hdmf/build/objectmapper.py

```diff
@@ -564,7 +565,11 @@ def get_attr_value(self, **kwargs):
                 msg = ("%s '%s' does not have attribute '%s' for mapping to spec: %s"
                        % (container.__class__.__name__, container.name, attr_name, spec))
                 raise ContainerConfigurationError(msg)
-            if attr_val is not None:
+            if isinstance(attr_val, TermSetWrapper):
+                # This is when the wrapped item is an attribute
+                # Refer to h5tools.py for wrapped datasets
```

Comment:
```suggestion
```
Actually this is used for some DatasetSpecs too (I think just those without a data_type). That is evidenced by the fact that this function was called when you changed the "my_data" DatasetSpec